tasks/facial_landmark_detection/search.py

```diff
@@ -0,0 +1,157 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+
+import sys
+# to be removed before merging
+if ('--debug' in sys.argv):
+    import debugpy
+    debugpy.listen(5678)
+    print("Waiting for debugger")
+    debugpy.wait_for_client()
+
+import math
+import os
+from argparse import ArgumentParser
+from datetime import datetime
+from pathlib import Path
+
+import pandas as pd
+import yaml
+from overrides.overrides import overrides
+
+from archai.common.common import logger
+from archai.discrete_search.algos.evolution_pareto import EvolutionParetoSearch
+from archai.discrete_search.api.model_evaluator import ModelEvaluator
+from archai.discrete_search.api.search_objectives import SearchObjectives
+from archai.discrete_search.evaluators.ray import RayParallelEvaluator
+
+import train as model_trainer
+from dataset import FaceLandmarkDataset
+from latency import AvgOnnxLatency
+from search_space import ConfigSearchSpaceExt
+class AccuracyEvaluator(ModelEvaluator):
+    
+    def __init__(self, lit_args) -> None:
+        self.lit_args = lit_args
+
+    @overrides
+    def evaluate(self, model, dataset_provider, budget = None) -> float:
+        logger.info(f"evaluating {model.arch.archid}")
+
+        val_error = model_trainer.train(self.lit_args, model.arch) #, model.arch.archid, dataset_provider)
+        if (math.isnan(val_error)):
+            logger.info(f"Warning: model {model.arch.archid} has val_error NaN. Set to 10000.0 to avoid corrupting the Pareto front.")
+            val_error = 10000.0
+        return val_error
+
+class OnnxLatencyEvaluator(ModelEvaluator):
+
+    def __init__(self, args) -> None:
+        self.args = args
+        self.latency_evaluator = AvgOnnxLatency(input_shape=(1, 3, 128, 128), num_trials=self.args.num_latency_measurements, num_input=self.args.num_input_per_latency_measurement)
+
+    @overrides
+    def evaluate(self, model, dataset_provider, budget = None) -> float:
+        return self.latency_evaluator.evaluate(model)
+        
+class SearchFaceLandmarkModels():
+    def __init__(self) -> None:
+        super().__init__()
+
+        config_parser = ArgumentParser(conflict_handler="resolve", description='NAS for Facial Landmark Detection.')
+        config_parser.add_argument("--config", required=True, type=Path, help='YAML config file specifying default arguments')
+        
+        parser = ArgumentParser(conflict_handler="resolve", description='NAS for Facial Landmark Detection.')
+        parser.add_argument("--output_dir", required=True, type=Path)
```

Comment:
For running this in AML it is handy if this script also takes the dataset dir as input rather than having it in the yaml file. 
This way the AML pipeline can pass in the same mounted blob store dataset folder form a prior dataset preparation component.  This mounted folder is then passed to the constructor of the FaceLandmarkDataset.